snapcraft/cli/lifecycle.py

```diff
@@ -180,13 +180,16 @@ def clean(parts, step, **kwargs):
     """
     project_options = get_project_options(**kwargs)
     build_environment = env.BuilderEnvironmentConfig()
-    if build_environment.is_host:
-        step = step or steps.next_step(step).name
+
+    if step:
```

Comment:
it gets confusing a bit when `step` can be a `step_name` or an actual instance of `Step`.

Reply:
Agreed, fixed by using an identifier decl for click to make it use a different parameter name.